packages/gatsby-transformer-remark/src/__tests__/extend-node.js

```diff
@@ -1248,24 +1248,65 @@ describe(`Headings are generated correctly from schema`, () => {
 ## second title
 `,
     `headings {
+      id
       value
       depth
     }`,
     node => {
       expect(node).toMatchSnapshot()
       expect(node.headings).toEqual([
         {
+          id: null,
           value: `first title`,
           depth: 1,
         },
         {
+          id: null,
           value: `second title`,
           depth: 2,
         },
       ])
     }
   )
 
+  bootstrapTest(
+    `returns value`,
+    `
+  # first title
+
+  ## second title
+  `,
+    `headings {
+        id
+        value
+        depth
+      }`,
+    node => {
+      expect(node).toMatchSnapshot()
+      expect(node.headings).toEqual([
+        {
+          id: `#first-title`,
```

Comment:
I think the `id` should be without leading `#` (alternatively the field should be named something else, because `#<id_text>` is not exactly `id` - it's "destination anchor" (at least that's the name used in https://www.w3.org/TR/REC-html40/struct/links.html )

Reply:
Yes you're right. It's without `#` when test the package locally.

I was trying to add a test with the plugin `gatsby-remark-autolink-headers` like:
```
  bootstrapTest(
    `returns id, value and depth`,
    `
  # first title

  ## second title
  `,
    `headings {
        id
        value
        depth
      }`,
    node => {
      expect(node).toMatchSnapshot()
      expect(node.headings).toEqual([
        {
          id: #first-title,
          value: `first title`,
          depth: 1,
        },
        {
          id: #second-title,
          value: `second title`,
          depth: 2,
        },
      ])
    },
    {
      pluginOptions: {
        plugins: [
          `gatsby-remark-autolink-headers`
        ],
      },
    }
  )
```
However, it failed to query with result like:
```
{
      errors: [
        GraphQLError {
          message: "Cannot read property 'name' of undefined",
          locations: [Array],
          path: [Array]
        }
      ],
      data: [Object: null prototype] { listNode: [ [Object: null prototype] ] }
}
```
I'm no clue from here :). Please let me know if you have any idea.

Reply:
I will check it out, I have no clues right now - I also saw the failing test, but just didn't have time to figure it out. The test setup seems fine but maybe theire is some trickery there